cmd/tailscale/cli/update.go

```diff
@@ -209,46 +221,24 @@ func (up *updater) updateSynology() error {
 func (up *updater) updateDebLike() error {
 	ver := updateArgs.version
 	if ver == "" {
-		res, err := http.Get("https://pkgs.tailscale.com/" + up.track + "/?mode=json")
+		var err error
+		ver, err = latestTailscaleVersion(up.track)
 		if err != nil {
 			return err
 		}
-		var latest struct {
-			Tarballs map[string]string // ~goarch (ignoring "geode") => "tailscale_1.34.2_mips.tgz"
-		}
-		err = json.NewDecoder(res.Body).Decode(&latest)
-		res.Body.Close()
-		if err != nil {
-			return fmt.Errorf("decoding JSON: %v: %w", res.Status, err)
-		}
-		f, ok := latest.Tarballs[runtime.GOARCH]
-		if !ok {
-			return fmt.Errorf("can't update architecture %q", runtime.GOARCH)
-		}
-		ver, _, ok = strings.Cut(strings.TrimPrefix(f, "tailscale_"), "_")
-		if !ok {
-			return fmt.Errorf("can't parse version from %q", f)
-		}
 	}
 	if up.currentOrDryRun(ver) {
 		return nil
 	}
 
-	track := "unstable"
-	if stable, ok := versionIsStable(ver); !ok {
-		return fmt.Errorf("malformed version %q", ver)
-	} else if stable {
-		track = "stable"
-	}
-
 	if os.Geteuid() != 0 {
 		return errors.New("must be root; use sudo")
```

Comment:
I keep thinking about this. It is bad to simply re-invoke ourselves with sudo?

And/or, should we re-tool the error message to talk about `doas` (OBSD at least, maybe others?) and `runas.exe` (Windows, when we get to that point)?

Reply:
Added a helper to check running as root and print an OS-appropriate message.
It doesn't look like `updateWindows` requires admin privileges, at least not explicitly in the code today.